lib/digital_signal/digital_signal.h

```diff
@@ -10,18 +10,35 @@
 extern "C" {
 #endif
 
-typedef struct {
+/* helper for easier signal generation */
+#define DIGITAL_SIGNAL_MS(x) (x * 100000000UL)
```

Comment:
<img width="790" alt="image" src="https://user-images.githubusercontent.com/1633132/236112143-4a0bf245-0190-474b-9c04-e5f1035e6feb.png">

Reply:
knowing it better and still having written this code, bugs me a bit :)
also thought i already fixed it somewhere.

Reply:
not sure why it thinks the expression in digital_signal.c:482 is always false.

the code caches dma_buffer->read_pos and checks for a change after calling digital_sequence_update_pos() which call updates dma_buffer->read_pos.
am i missing something?

![grafik](https://user-images.githubusercontent.com/479231/236131025-43aa6d07-86d5-456b-bd73-e5dd35fa9c8a.png)